src/susceptibility.cpp

```diff
@@ -431,13 +431,6 @@ void *gyrotropic_susceptibility::copy_internal_data(void *data) const {
   return (void *)dnew;
 }
 
-bool gyrotropic_susceptibility::needs_P(component c, int cmp,
-                                        realnum *W[NUM_FIELD_COMPONENTS][2]) const {
-  if (!is_electric(c) && !is_magnetic(c)) return false;
-  direction d0 = component_direction(c);
-  return (d0 == X || d0 == Y || d0 == Z) && sigma[c][d0] && W[c][cmp];
-}
-
```

Comment:
It seems like we may be better off keeping this simpler `needs_P` criterion, and changing `gyrotropic_susceptibility::num_cinternal_notowned_needed` to return `0`.

Reason: since no offdiagonal averaging is done for the P components (https://github.com/NanoComp/meep/issues/1117#issuecomment-628591902), we never need to communicate the "not owned" P grid points from neighboring chunks.   So, we only need to allocate P if *this* chunk has the corresponding sigma.

Reply:
Yes, it seems that this is the correct fix (6310d75).

I verified that the Faraday rotation example still works, and the test case from @mojv no longer crashes.